homeassistant/components/prometheus/__init__.py

```diff
@@ -103,6 +103,16 @@ def _metric(self, metric, factory, documentation, labels=None):
                 full_metric_name, documentation, labels)
             return self._metrics[metric]
 
+    @staticmethod
+    def state_as_number(state):
+        """Return a state casted to a float."""
+        try:
+            value = state_helper.state_as_number(state)
+        except ValueError:
+            _LOGGER.warning("Could not convert %s to float", state)
```

Comment:
Will there be complaints about the warning? We could lower it to debug, although I think that warning is more appropriate.

Reply:
@tcastberg thoughts?

Reply:
I could go either way. I'm not aware of which rules (if any) the project has for logging level detail. I'd rather have it higher than lower since I'm not sure this is a permanent fix. A more appropriate, long-term solution would automatically generate permanent, unique id's for different zones somewhere in HA. I was more worried about doing what was achievable to reduce the amount of exception spam in the logs, but I do want this to warn that it is still an issue.